nucliadb/nucliadb/train/generator.py

```diff
@@ -56,54 +72,50 @@ async def generate_train_data(kbid: str, shard: str, trainset: TrainSet):
                 detail="Paragraph Classification should be of 1 labelset",
             )
 
-        async for paragraph_data in generate_paragraph_classification_payloads(
+        payloads_generator = generate_paragraph_classification_payloads(
             kbid, trainset, node, shard_replica_id
-        ):
-            payload = paragraph_data.SerializeToString()
-            yield len(payload).to_bytes(4, byteorder="big", signed=False)
-            yield payload
+        )
 
-    if trainset.type == TaskType.FIELD_CLASSIFICATION:
+    elif trainset.type == TaskType.FIELD_CLASSIFICATION:
         if len(trainset.filter.labels) != 1:
             raise HTTPException(
                 status_code=422,
                 detail="Field Classification should be of 1 labelset",
             )
 
-        async for field_data in generate_field_classification_payloads(
+        payloads_generator = generate_field_classification_payloads(
             kbid, trainset, node, shard_replica_id
-        ):
-            payload = field_data.SerializeToString()
-            yield len(payload).to_bytes(4, byteorder="big", signed=False)
-            yield payload
+        )
 
-    if trainset.type == TaskType.TOKEN_CLASSIFICATION:
-        async for token_data in generate_token_classification_payloads(
+    elif trainset.type == TaskType.TOKEN_CLASSIFICATION:
+        payloads_generator = generate_token_classification_payloads(
             kbid, trainset, node, shard_replica_id
-        ):
-            payload = token_data.SerializeToString()
-            yield len(payload).to_bytes(4, byteorder="big", signed=False)
-            yield payload
+        )
 
-    if trainset.type == TaskType.SENTENCE_CLASSIFICATION:
+    elif trainset.type == TaskType.SENTENCE_CLASSIFICATION:
         if len(trainset.filter.labels) == 0:
             raise HTTPException(
                 status_code=422,
                 detail="Sentence Classification should be at least of 1 labelset",
             )
 
-        async for sentence_data in generate_sentence_classification_payloads(
+        payloads_generator = generate_sentence_classification_payloads(
             kbid, trainset, node, shard_replica_id
-        ):
-            payload = sentence_data.SerializeToString()
-            yield len(payload).to_bytes(4, byteorder="big", signed=False)
-            yield payload
+        )
 
-    if trainset.type == TaskType.IMAGE_CLASSIFICATION:
-        async for image_data in generate_image_classification_payloads(
+    elif trainset.type == TaskType.IMAGE_CLASSIFICATION:
+        payloads_generator = generate_image_classification_payloads(
             kbid, trainset, node, shard_replica_id
-        ):
-            payload = image_data.SerializeToString()
+        )
+
+    elif trainset.type == TaskType.PARAGRAPH_STREAMING:
+        payloads_generator = generate_paragraph_streaming_payloads(
+            kbid, trainset, node, shard_replica_id
+        )
+
```

Comment:
nit: should we handle the `else` branch?

Reply:
Sure, I can return a 400 saying it's an invalid training